### DISCUSSION TITLE
Adding default node selectors for dragonfly instances

### DISCUSSION THREAD

**Discussion by danfinn** [dragonflydb/dragonfly#2275] [General]
I noticed that you apply a nodeselector of kubernetes.io/os = linux to the operator but you don't apply the same to dragonfly instances.  This quickly caused an issue for me because (unfortunately) we run our clusters with both linux and windows node pools.  Could/should you also apply the same linux node selector to dragonfly instances?

**Comment by romange** [comment]
would you like to submit a pr?

**Reply by danfinn** [reply]
I could try.  I'm not super familiar with CRDs but I could give it a go.  I actually expected you to respond with a reason why this isn't being done that I was missing.

Would this be something that gets set in the CRD or somewhere else?

**Reply by romange** [reply]
@Pothulapati can answer this - he wrote the operator.

**Reply by markusbr** [reply]
@romange @Pothulapati can you please advise on how to pass down a node selector like `cloud.google.com/gke-spot: "true"` to the Kubernetes operator in order to use preemptible instances for the dragonfly instances?

**Reply by Pothulapati** [reply]
Okay, My bad for missing this. 

For Operator, This is pretty straightforward. We have the affinity field https://github.com/dragonflydb/dragonfly-operator/blob/main/config/crd/bases/dragonflydb.io_dragonflies.yaml#L56 through which you can pass down configurations like this. https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
